packages/backend/src/scheduler/SchedulerTask.ts

```diff
@@ -365,6 +368,39 @@ export const sendSlackNotification = async (
     }
 };
 
+export const downloadCsv = async (
+    jobId: string,
+    scheduledTime: Date,
+    payload: DownloadCsvPayload,
+) => {
+    const baseLog: Pick<SchedulerLog, 'task' | 'jobId' | 'scheduledTime'> = {
+        task: 'downloadCsv',
+        jobId,
+        scheduledTime,
+    };
+    try {
+        schedulerService.logSchedulerJob({
+            ...baseLog,
+            details: { createdByUserUuid: payload.userUuid },
+            status: SchedulerJobStatus.STARTED,
+        });
+
+        const fileUrl = await csvService.downloadCsv(jobId, payload);
+        schedulerService.logSchedulerJob({
+            ...baseLog,
+            details: { fileUrl, createdByUserUuid: payload.userUuid },
+            status: SchedulerJobStatus.COMPLETED,
+        });
+    } catch (e) {
+        schedulerService.logSchedulerJob({
+            ...baseLog,
+            status: SchedulerJobStatus.ERROR,
+            details: { createdByUserUuid: payload.userUuid, error: e },
+        });
+        // do not throw error to avoid retrying
```

Comment:
I assumed since you scheduled the job with only 1 retry that even if an error is thrown, it wouldn't retry? Or am I misunderstanding?

Could you explain the retry logic?

Reply:
tasks can be retried if an exception is triggered. I'm not triggering any exception (captured by catch) here, so i twill not be retried. 

But you're right that when I create this task I set `maxAttempts: 1` so it will not be retried even if an exception is thrown... 

I guess it is a bit redundant, but I feel the comment might help in the future if we want to make it retry.